cmd/query/app/querysvc/adjuster/ipattribute.go

```diff
@@ -0,0 +1,72 @@
+package adjuster
+
+import (
+	"bytes"
+	"encoding/binary"
+	"strconv"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+var ipAttributesToCorrect = map[string]struct{}{
+	"ip":        {},
+	"peer.ipv4": {},
+}
+
+// IPAttribute returns an adjuster that replaces numeric "ip" attributes,
+// which usually contain IPv4 packed into uint32, with their string
+// representation (e.g. "8.8.8.8"").
+func IPAttribute() Adjuster {
+	adjustAttributes := func(attributes pcommon.Map) {
+		adjusted := make(map[string]string)
+
+		attributes.Range(func(k string, v pcommon.Value) bool {
+			if _, ok := ipAttributesToCorrect[k]; !ok {
+				return true
+			}
+			var value uint32
+			switch v.Type() {
+			case pcommon.ValueTypeInt:
+				//nolint: gosec // G115
+				value = uint32(v.Int())
+			case pcommon.ValueTypeDouble:
+				value = uint32(v.Double())
+			default:
+				return true
+			}
+			var buf [4]byte
+			binary.BigEndian.PutUint32(buf[:], value)
+			var sBuf bytes.Buffer
+			for i, b := range buf {
+				if i > 0 {
+					sBuf.WriteRune('.')
+				}
+				sBuf.WriteString(strconv.FormatUint(uint64(b), 10))
+			}
+			adjusted[k] = sBuf.String()
+			return true
+		})
+
+		for k, v := range adjusted {
+			attributes.PutStr(k, v)
+		}
+	}
+
+	return Func(func(traces ptrace.Traces) (ptrace.Traces, error) {
```

Comment:
@yurishkuro The function in v1 looks like this 
```go
	return Func(func(trace *model.Trace) (*model.Trace, error) {
		for _, span := range trace.Spans {
			adjustTags(span.Tags)
			adjustTags(span.Process.Tags)
			model.KeyValues(span.Process.Tags).Sort()
		}
		return trace, nil
	})
```
Couple of questions: 
- Are process tags also just part of attributes in OTLP?
- Do these attributes need to be sorted?

Reply:
* process tags are translated into Resource.attributes
* I don't think they need to be sorted here.